src/api/endpoints/track.ts

```diff
@@ -0,0 +1,85 @@
+import { apiClient } from "../client.ts";
+import { handleApiError } from "../../utils.ts";
+import type { AxiosResponse } from "axios";
+import type {
+  CreateTrackPayload,
+  ListTrackParams,
+  Track,
+  TrackListResponse,
+  UpdateTrackPayload,
+  UploadTrackPayload,
+} from "../types/track.ts";
+
+const RESOURCE = "/api/tracks";
+
+export async function createTrack(data: CreateTrackPayload, opts = {}) {
+  try {
+    const response = await apiClient.post<
+      Track,
+      AxiosResponse<Track>,
+      CreateTrackPayload
+    >(RESOURCE, data, opts);
+    return response.data;
+  } catch (e) {
+    handleApiError(e);
+  }
+}
+
+export async function updateTrack(id: Track["id"], data: UpdateTrackPayload) {
+  try {
+    const response = await apiClient.put<
+      Track,
+      AxiosResponse<Track>,
+      UpdateTrackPayload
+    >(`${RESOURCE}/${id}`, data);
+    return response.data;
+  } catch (e) {
+    handleApiError(e);
+  }
+}
+
+export async function uploadTrack(id: Track["id"], data: UploadTrackPayload) {
+  try {
+    const response = await apiClient.put<
+      Track,
+      AxiosResponse<Track>,
+      UploadTrackPayload
+    >(`${RESOURCE}/${id}/upload`, data, {
+      headers: {
+        "Content-Type": "multipart/form-data",
+      },
+    });
+    return response.data;
+  } catch (e) {
+    handleApiError(e);
+  }
+}
+
+export async function deleteTrack(id: Track["id"]) {
+  try {
+    const response = await apiClient.delete<void>(`${RESOURCE}/${id}`);
+    return response.data;
+  } catch (e) {
+    handleApiError(e);
+  }
+}
+
+export async function loadTrackBySlug(slug: Track["slug"]) {
+  try {
+    const response = await apiClient.get<Track>(RESOURCE, { params: { slug } });
```

Comment:
When I reload page from edit track modal(which is by the way very cool, that even this state is a separate link) http://localhost:3000/tracks/{some_track}/edit, then I get:
![image](https://github.com/user-attachments/assets/e4761e30-4eeb-4a87-bf57-f2311bd39dc1)

instead of:
![image](https://github.com/user-attachments/assets/8f8ed8a6-d865-40c9-b08c-d652690ecbf2)


```suggestion
    const response = await apiClient.get<Track>(`${RESOURCE}/${slug}`);
```

Reply:
Good catch, fixed